L1Trigger/TrackFindingTracklet/interface/Util.h

```diff
@@ -35,150 +39,142 @@ namespace trklet {
     return str;
   }
 
-  //Should be optimized by layer - now first implementation to make sure it works OK
-  inline int bendencode(double bend, bool isPS) {
-    int ibend = 2.0 * bend;
+  inline double bendstrip(double r, double rinv, double stripPitch) {
+    constexpr double dr = 0.18;
+    double delta = r * dr * 0.5 * rinv;
+    double bend = delta / stripPitch;
+    return bend;
+  }
 
-    assert(std::abs(ibend - 2.0 * bend) < 0.1);
+  inline double rinv(double phi1, double phi2, double r1, double r2) {
+    if (r2 <= r1) {  //FIXME can not form tracklet should not call function with r2<=r1
+      return 20.0;
+    }
 
-    if (isPS) {
-      if (ibend == 0 || ibend == 1)
-        return 0;
-      if (ibend == 2 || ibend == 3)
-        return 1;
-      if (ibend == 4 || ibend == 5)
-        return 2;
-      if (ibend >= 6)
-        return 3;
-      if (ibend == -1 || ibend == -2)
-        return 4;
-      if (ibend == -3 || ibend == -4)
-        return 5;
-      if (ibend == -5 || ibend == -6)
-        return 6;
-      if (ibend <= -7)
-        return 7;
-
-      throw cms::Exception("BadConfig") << __FILE__ << " " << __LINE__
-                                        << " Unknown bendencode for PS module for bend = " << bend
-                                        << " ibend = " << ibend;
+    double dphi = phi2 - phi1;
+    double dr = r2 - r1;
+
+    return 2.0 * sin(dphi) / dr / sqrt(1.0 + 2 * r1 * r2 * (1.0 - cos(dphi)) / (dr * dr));
+  }
+
+  inline std::string convertHexToBin(const std::string& stubwordhex) {
+    std::string stubwordbin = "";
+
+    for (char word : stubwordhex) {
+      std::string hexword = "";
+      if (word == '0')
```

Comment:
Do you expect a comparison to happen once per iteration? If yes, why not use else if? It seems with this way, the code will compare all which is not necessary.

Reply:
Yes, this should probably be a else if...

Reply:
changed to else ifs